src/platform/unix/mod.rs

```diff
@@ -962,6 +960,20 @@ fn create_shmem(name: CString, length: usize) -> c_int {
     }
 }
 
+#[cfg(target_os="freebsd")]
+fn create_shmem(_name: CString, length: usize) -> c_int {
```

Comment:
This is a bit dirty: since the name construction has side effects, I doubt the compiler is smart enough to realise it can be omitted altogether...

Though to be honest, I'm not at all sure whether it's really worthwhile to refactor the code in order to avoid this -- so I guess that's just me thinking out loud, rather than an actual request to change anything :-)